### ISSUE TITLE
Get-AtwsContact Change -> Get-AtwsData : Cannot bind argument to parameter 'Filter' because it is an empty string.

### ISSUE THREAD

**Issue by rilesthegiles** [ecitsolutions/Autotask#73]
```
At [Censored]\Dynamic\Get-AtwsContact.ps1:732 char:64
+ ...          $result = Get-AtwsData -Entity $entityName -Filter $Filter `
+                                                                 ~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-AtwsData], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Get-AtwsData
```
Need parameter filtering to pass through empty strings. 

Currently when using `Get-AtwsContact -ParameterName $Parameter` if `$null -eq $Parameter` it will throw this error. However, after throwing this error `Get-AtwsContact` will return an object with the following information:
```
$u = Get-AtwsContact -ParameterName $null
$u.GetType()
<### Results ###>
IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     Resource                                 Autotask.Entity
<############>
$u
<### Results ###>
EmailTypeCodeLabel         : Primary e-mail address
EmailTypeCode2Label        :
EmailTypeCode3Label        :
GenderLabel                :
GreetingLabel              :
LocationIDLabel            : Censored
ResourceTypeLabel          : Employee
SuffixLabel                :
TravelAvailabilityPctLabel :
UserTypeLabel              : API User (system)
DateFormatLabel            : MM/dd/yyyy
TimeFormatLabel            : hh:mm a
PayrollTypeLabel           : Salary
NumberFormatLabel          : X,XXX.XX
LicenseTypeLabel           : API User
Active                     : True
Email                      : Censored
Email2                     :
Email3                     :
EmailTypeCode              : PRIMARY
EmailTypeCode2             :
EmailTypeCode3             :
FirstName                  : Censored
Gender                     :
Greeting                   :
HomePhone                  :
Initials                   :
LastName                   : API
LocationID                 : 
MiddleName                 :
MobilePhone                :
OfficeExtension            :
OfficePhone                :
ResourceType               : Employee
Suffix                     :
Title                      :
TravelAvailabilityPct      :
UserName                   : Censored
UserType                   : 13
DefaultServiceDeskRoleID   :
DateFormat                 : MM/dd/yyyy
TimeFormat                 : hh:mm a
Password                   : ******
PayrollType                : Censored
NumberFormat               : X,XXX.XX
AccountingReferenceID      :
InternalCost               : 0.0000
HireDate                   : Censored
SurveyResourceRating       :
LicenseType                : Censored
Fields                     :
id                         : Censored
UserDefinedFields          : {}
<############>
```
My expected result is if `$Parameter = $null`, then take out `ParameterName` from the `Get-AtwsData` request. If there are no other ParameterNames, then throw an error. Otherwise, search utilizing the other `ParameterName`'s.

**Comment by rilesthegiles** [comment]
Also I haven't had much development experience on GIT but I maybe able to fix this issue over weekend and have one of the contributors of this GIT look over it, however that process works :)

**Comment by klemmestad** [comment]
Hi, I can see that this is a weird and unexpected result. I discovered it during debugging something else a couple of weeks ago. The fix is included in the just released 1.6.5 version. It is an error of variable scope at my part. When you connect to Autotask I use a Get-AtwsResource on the current username to verify that the connection is really working. I just didn't realise that using the variable name `$result` at that point would provide a dummy value for later, failed queries...

With regards to the expected result I see it differently. What I expect - and admittedly, has coded - is the equivalent of using `Get-AtwsContact -IsNull ParameterName`, i.e. searching for contacts where the parameter has the value `$null`. I do believe this is more in accordance with how the rest of the module behaves.

**Comment by rilesthegiles** [comment]
Great to hear man thanks! Will cut down on the coding for my side with
output validation. Your module has been a life saver for me at my work. I'm
currently building out a script to compile email chains into their same
ticket across companies, since AutoTask will create a new ticket if it's a
different company.
